src/components/RequireSubmission.js

```diff
@@ -10,11 +13,23 @@ import { Redirect } from 'react-router-dom';
  * If there is no submission, the user is forcibly redirected to the start of the form.
  */
 const RequireSubmission = ({ submission, component: Component, ...props }) => {
+  const maintenanceMode = props?.form?.maintenanceMode;
+  const isFormDesigner = IsFormDesignerHeader.getValue();
+  let maintenanceModeAlert = null;
+  if (!isFormDesigner && maintenanceMode) {
+    throw new ServiceUnavailable('Service Unavailable', 503, 'Form in maintenance', 'form-maintenance');
+  } else if (isFormDesigner && maintenanceMode) {
+    maintenanceModeAlert = <MaintenanceMode />;
+  }
+
```

Comment:
Wouldn't it make more sense to do this after the check for whether there is a submission?

Reply:
It was originally like that, but then the page rendered the form step briefly (half a second or so), to then throw and go into the error boundary, which was a really janky user experience.

In terms of UX, if you can detect an error situation before the user even does any interaction (explicit or implicit), you want to handle it asap.